### ISSUE TITLE
Sampling from joint normal distribution fails

### ISSUE THREAD

**Issue by rikardn** [sympy/sympy#17622]
The following code that would sample from a joint normal distribution fails:

```python
import sympy.stats
X = sympy.stats.Normal('X', [0, 0], [[0.1, 0.025], [0.025, 0.1]])
sympy.stats.sample(X)
```

The latter part of the error message:
```
  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 78, in domain
    return SingleDomain(self.symbol, self.set)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 50, in set
    return self.domain.set

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv.py", line 76, in domain
    rvs = random_symbols(self.distribution)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/rv.py", line 510, in random_symbols
    return list(atoms(RandomSymbol))

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/core/basic.py", line 507, in atoms
    for expr in preorder_traversal(self):

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/core/basic.py", line 2007, in __next__
    return next(self._pt)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/core/basic.py", line 1978, in _preorder_traversal
    for subtree in self._preorder_traversal(arg, keys):

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/core/basic.py", line 1978, in _preorder_traversal
    for subtree in self._preorder_traversal(arg, keys):

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/core/basic.py", line 1978, in _preorder_traversal
    for subtree in self._preorder_traversal(arg, keys):

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/core/basic.py", line 1965, in _preorder_traversal
    if isinstance(node, Basic):

RecursionError: maximum recursion depth exceeded while calling a Python object
```

I am using sympy 1.4 and python 3.7.3

**Comment by rikardn** [comment]
I get the same error with sympy 1.5

**Comment by czgdp1807** [comment]
Still not fixed.

**Comment by Smit-create** [comment]
Is this issue related to `lambdify` also?
```py
>>> from sympy.stats import MultivariateNormal
>>> from sympy import lambdify
>>> m = MultivariateNormal('X', [1, 2], [[2, 1], [1, 2]])
>>> list(m.pspace.values)
[X]
>>> m
X
>>> fn = lambdify(list(m.pspace.values)[0], m)
...

File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1233, in _imp_namespace
    _imp_namespace(arg, namespace)
  File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1255, in _imp_namespace
    _imp_namespace(arg, namespace)
  File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1233, in _imp_namespace
    _imp_namespace(arg, namespace)
  File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1255, in _imp_namespace
    _imp_namespace(arg, namespace)
 File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1233, in _imp_namespace
    _imp_namespace(arg, namespace)
  File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1255, in _imp_namespace
    _imp_namespace(arg, namespace)
  File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1233, in _imp_namespace
    _imp_namespace(arg, namespace)
  File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1255, in _imp_namespace
    _imp_namespace(arg, namespace)
  File "/home/smit/Smitlunagariya/sympy/sympy/utilities/lambdify.py", line 1232, in _imp_namespace
    for arg in expr:
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/joint_rv.py", line 189, in __getitem__
    if (self.pspace.component_count <= key) == True:
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/joint_rv.py", line 66, in component_count
    _set = self.distribution.set
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/joint_rv_types.py", line 135, in set
    return S.Reals**k
  File "/home/smit/Smitlunagariya/sympy/sympy/core/decorators.py", line 251, in _func
    return func(self, other)
  File "/home/smit/Smitlunagariya/sympy/sympy/sets/sets.py", line 660, in __pow__
    if not (exp.is_Integer and exp >= 0):
  File "/home/smit/Smitlunagariya/sympy/sympy/core/numbers.py", line 2274, in __ge__
    other = _sympify(other)
  File "/home/smit/Smitlunagariya/sympy/sympy/core/sympify.py", line 498, in _sympify
    return sympify(a, strict=True)
  File "/home/smit/Smitlunagariya/sympy/sympy/core/sympify.py", line 347, in sympify
    return conv(a)
  File "/home/smit/Smitlunagariya/sympy/sympy/core/cache.py", line 72, in wrapper
    retval = cfunc(*args, **kwargs)
RecursionError: maximum recursion depth exceeded while calling a Python object

```
The error is because of:
https://github.com/sympy/sympy/blob/1a62f56afdb6421d221c7ca07942f7ac351e399a/sympy/stats/rv.py#L1160-L1161
This needs to be fixed in order to pass on the call to JointPSpace.sample() to return the sample

**Comment by rikardn** [comment]
I can confirm that this now works in the 1.7 development version. The fix didn't make it into 1.6.2 in case anyone waits for the fix.
